File: chapter4/typescript/PotatoChip.ts
import { Product } from "./Product";

/**
 * PotatoChip product information
 *
 * @Author Bridget Wu
 */
 export class PotatoChip implements Product {
    private id: number = 2;

    private name: String = "Potato Chip";

    private price: number =0 ;

    public constructor(price: number) {
        this.price = price;
    }

    public getId(): number {
        return this.id;
    }

    public getPrice(): number {
        return this.price;
    }

    public getName(): String {
        return this.name;
    }
}

File: chapter4/java/designpattern/src/main/java/com/designpattern/PotatoChip.java
package com.designpattern;

/**
 * PotatoChip product information
 *
 * @Author Bridget Wu
 */
public class PotatoChip implements Product {
    private int id = 2;

    private String name = "Potato Chip";

    private int price;

    public PotatoChip(int price) {
        this.price = price;
    }


    @Override
    public int getId() {
        return this.id;
    }

    @Override
    public int getPrice() {
        return this.price;
    }

    @Override
    public String getName() {
        return this.name;
    }
}

